src/PhpSessionPersistence.php

```diff
@@ -66,7 +93,21 @@ public function persistSession(SessionInterface $session, ResponseInterface $res
                 ->withValue(session_id())
                 ->withPath(ini_get('session.cookie_path'));
 
-            return FigResponseCookies::set($response, $sessionCookie);
+            $response = FigResponseCookies::set($response, $sessionCookie);
+
+            if ($this->cacheLimiter) {
+                if ($this->responseAlreadyHasCacheHeaders($response)) {
+                    return $response;
+                }
+                $cacheHeaders = $this->generateCacheHeaders($this->cacheLimiter, $this->cacheExpire);
+                foreach ($cacheHeaders as $name => $value) {
+                    if (false !== $value) {
+                        $response = $response->withHeader($name, $value);
+                    }
+                }
+            }
+
+            return $response;
```

Comment:
Invert the conditional so you can return early, combine conditions, and promote the body of the conditional:

```php
if (! $this->cacheLimiter || $this->responseAlreadyHasCacheHeaders($response)) {
    return $response;
}

$cacheHeaders = $this->generateCacheHeaders($this->cacheLimiter, $this->cacheExpire);
// ...
return $response;
```